src/main/java/com/flippingutilities/ui/login/LoginPanel.java

```diff
@@ -0,0 +1,318 @@
+package com.flippingutilities.ui.login;
+
+import com.flippingutilities.controller.FlippingPlugin;
+import com.flippingutilities.ui.uiutilities.CustomColors;
+import com.flippingutilities.ui.uiutilities.Icons;
+import com.flippingutilities.ui.uiutilities.UIUtilities;
+import net.runelite.client.ui.ColorScheme;
+import net.runelite.client.ui.DynamicGridLayout;
+import net.runelite.client.ui.components.IconTextField;
+
+import javax.swing.*;
+import javax.swing.border.CompoundBorder;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.MatteBorder;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.time.Instant;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class LoginPanel extends JPanel{
+    FlippingPlugin plugin;
+    Runnable onViewChange;
+    public boolean showingLoggedInView;
+    JLabel slotFeatureHealthLabel = new JLabel("Inactive (not logged in)");
+    Instant timeOfLastSuccessfulRequest;
+    boolean errorWhenSendingSlotRequest;
+    boolean notSendingRequestDueToNoChange;
+
+    public LoginPanel(FlippingPlugin plugin) {
+        this.plugin = plugin;
+        plugin.getApiAuthHandler().subscribeToLogin(this::showLoggedInView);
+        add(createLoggedOutPanel());
+        Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(this::checkHealth, 5, 1, TimeUnit.SECONDS);
+    }
+
+    public void addOnViewChange(Runnable r) {
+        this.onViewChange = r;
+    }
+
+    private void checkHealth() {
+        SwingUtilities.invokeLater(() -> {
+            if (plugin.getCurrentlyLoggedInAccount() == null) {
+                timeOfLastSuccessfulRequest = null;
+                slotFeatureHealthLabel.setText("Inactive (not logged in)");
+            }
+            else if (!plugin.getApiAuthHandler().canCommunicateWithApi(plugin.getCurrentlyLoggedInAccount())) {
+                slotFeatureHealthLabel.setText("Inactive (not registered yet)");
+            }
+            else if (timeOfLastSuccessfulRequest == null) {
+                slotFeatureHealthLabel.setText("Active (starting up)");
+            }
+            else if (notSendingRequestDueToNoChange) {
+                slotFeatureHealthLabel.setText("Active (no slot change detected)");
+            }
+            else if (errorWhenSendingSlotRequest) {
+                slotFeatureHealthLabel.setText("Inactive (error on slot update)");
+            }
+            else {
+                long diff = Instant.now().getEpochSecond() - timeOfLastSuccessfulRequest.getEpochSecond();
+                slotFeatureHealthLabel.setText(String.format("Active (%ds since last update)", diff));
+            }
+            revalidate();
+            repaint();
+        });
+    }
+
+    public void onSlotRequest(int success) {
```

Comment:
int is a bit confusing if we don't know the context, an enum might be worth using